test/test_add_products.py

```diff
@@ -0,0 +1,22 @@
+import pytest
+from fastapi.testclient import TestClient
+from main import app
+from src.models import ProductCreate
+
+client = TestClient(app)
+
+
+@pytest.fixture
+def product_data():
+    return {
+        'name': 'Product Name',
+        'price': 9.99,
+        'type': 'food',
+        'region': 'Europe'
+    }
+
+
+def test_add_product_to_db(product_data):
+    response = client.post('/products', json=product_data)
+    assert response.status_code == 201
+    assert response.json() == product_data
```

Comment:
Just one test? What if you pass the product when

* Price is negative
* Name is empty
* Type is not one of possible variants

?

Reply:
All of possible use cases must be covered with tests